fetch/src/2_process.r

```diff
@@ -0,0 +1,36 @@
+process_data <- function() {
+  #Grab the data
+  mendota_file <- file.path(project_output_dir, 'model_RMSEs.csv')
+  # Prepare the data for plotting
+  eval_data <- readr::read_csv(mendota_file, col_types = 'iccd') %>%
+    filter(str_detect(exper_id, 'similar_[0-9]+')) %>%
+    mutate(col = case_when(
+      model_type == 'pb' ~ '#1b9e77',
+      model_type == 'dl' ~'#d95f02',
+      model_type == 'pgdl' ~ '#7570b3'
+    ), pch = case_when(
+      model_type == 'pb' ~ 21,
+      model_type == 'dl' ~ 22,
+      model_type == 'pgdl' ~ 23
+    ), n_prof = as.numeric(str_extract(exper_id, '[0-9]+')))
+  # Save the processed data
+  readr::write_csv(eval_data, file = file.path(project_output_dir, 'model_summary_results.csv'))
+  # Save the model diagnostics
```

Comment:
One aspect of a targets-oriented pipelining approach that you'll get more familiar with later in this course is that we typically try to modularize our functions so that each function only produces a single output, whether that's a file or an object like a list or dataframe. So here, even though the data processing, data saving, and the data rendering steps do fall into the `2_process` phase, we'd actually split them into separate functions, one which processes the data, one that saves it as a csv file (though that step could be incorporated into the first if desired), and a final function that reads in that processed data, renders the diagnostics, and saves the diagnostics to a text file. Since your code runs fine as is, there's no need to make the changes now (unless you'd like to). In an upcoming issue you'll be reworking the code to fit into a `targets` framework, so will have a chance to modularize this more then.

Reply:
And for pipelines, it's just fine to have multiple functions defined in a single script, so no need to make additional .R files in `2_process/src`, though we do occasionally do so for big projects where there are many different 2_process tasks.

Reply:
Oh okay, I see.